Batteries/Util/LibraryNote.lean

```diff
@@ -35,6 +35,45 @@ creates a new "library note", which can then be cross-referenced using
 -- See note [some tag]
 ```
 in doc-comments.
+Use `#help note "some tag"` to display all notes with the tag
+`"some tag"` in the infoview.
 -/
 elab "library_note " title:strLit ppSpace text:docComment : command => do
   modifyEnv (libraryNoteExt.addEntry · (title.getString, text.getDocString))
+
+
+open Lean Parser in
+/--
+`#help note "foo"` searches (case-insensitively) for all library notes whose
+label starts with "foo", then displays those library notes sorted alphabetically by label,
+grouped (case-sensitively) by label.
+The command only displays the library notes that are declared in
+imported files or in the same file above the line containing the command.
+-/
+elab "#help note" name:strLit : command => do
+  let env ← getEnv
+
+  -- get the library notes from both this and imported files
+  let local_entries := (libraryNoteExt.getEntries env).reverse
+  let imported_entries := (libraryNoteExt.toEnvExtension.getState env).importedEntries
+
+  -- filter for the appropriate notes while casting to list
+  let label_prefix := name.getString.toLower
```

Comment:
Are tags really case insensitive?

Reply:
no, but i think it's useful to search case-insensitively. needing to remember the exact casing of a label seems like something we can prevent

Reply:
aha... so `String.toLower` only works on basic latin characters... that's good to know. I'll think about making the a PR addressing the insensitivity.